{"uid":"3f804b9397f2e684","name":"test_to_verify_digital_card_creation_in_bulk_with_same_mob_with_negative_scenario","fullName":"TestCases.gcBulk.test_fino_digital_bulk_gc_card_creation.TestFinoDigitalBulkGiftCardCreation#test_to_verify_digital_card_creation_in_bulk_with_same_mob_with_negative_scenario","historyId":"1cc1040ad467c3c58f801699da63ddb7","time":{"start":1738695199677,"stop":1738695320618,"duration":120941},"description":"Verify the bulk creation of digital gift cards with the same mobile number in a negative scenario.","descriptionHtml":"<p>Verify the bulk creation of digital gift cards with the same mobile number in a negative scenario.</p>\n","status":"failed","statusMessage":"AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.","statusTrace":"self = <TestCases.gcBulk.test_fino_digital_bulk_gc_card_creation.TestFinoDigitalBulkGiftCardCreation object at 0x102b089b0>\nfinoObject = <constant.fino.fino_constant.FinoConstant object at 0x102cd92e0>\n\n    @allure.description(\n        \"Verify the bulk creation of digital gift cards with the same mobile number in a negative scenario.\")\n    def test_to_verify_digital_card_creation_in_bulk_with_same_mob_with_negative_scenario(self, finoObject):\n>       DigitalGiftCardCreationInBulk.verify_zaggle_digital_card_creation_in_bulk_with_same_mob_negative_scenario(\n            finoObject.FINO_GC_CARD_PROGRAM_ID, \"fino_same_mob_neg\", 50, finoObject.ADMIN_LOGIN_TOKEN, Bank.FINO\n        )\n\nTestCases/gcBulk/test_fino_digital_bulk_gc_card_creation.py:101: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages/gcBulk/digitalGc/DigitalCardCreationInBulk.py:365: in verify_zaggle_digital_card_creation_in_bulk_with_same_mob_negative_scenario\n    DigitalGiftCardCreationInBulk.verify_fino_failed_bulk_card_creation_for_same_mob_number(response,\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nresponse = <Response [200]>, records = 50, token = 'rYkNz0AQn4nyMMjDGw33T9pliEE5I7tmfKnow5L2bJ8='\n\n    @staticmethod\n    @allure.step(\"verify failed bulk upload for same mob number \")\n    def verify_fino_failed_bulk_card_creation_for_same_mob_number(response, records, token):\n        with allure.step(\"Verify bulk upload constraints and validate results\"):\n            CommonMethod.timeTakenForBulkUpload(records)\n    \n            bulkUploadId = response.json().get(\"bulkUploadId\")\n            allure.attach(bulkUploadId, name=\"Bulk Upload ID\", attachment_type=allure.attachment_type.TEXT)\n    \n            responseOfBulkUpload = BulkUploadDetails.get_bulk_upload_details(bulkUploadId, token)\n            data = responseOfBulkUpload.json()\n            allure.attach(str(data), name=\"Bulk Upload Details\", attachment_type=allure.attachment_type.JSON)\n    \n            status = data.get(\"status\")\n    \n            # assert status.lower() == \"failed\", \"Bulk upload should fail,whole file will failed for zaggle\"\n    \n            # else:\n            assert status.lower() == \"completed\", \"As 30 card will created successfully with same mob so status will be completed, This is for fino rest of all failed\"\n    \n            total_card_requests = data.get(\"totalCardsRequested\")\n            cards_not_created = data.get(\"cardsNotCreated\")\n            cards_created_successfully = data.get(\"cardsCreatedSuccessfully\")\n            allure.attach(f\"Total Cards Requested: {total_card_requests}\", name=\"Total Cards Requested\",\n                          attachment_type=allure.attachment_type.TEXT)\n            allure.attach(f\"Cards Not Created: {cards_not_created}\", name=\"Cards Not Created\",\n                          attachment_type=allure.attachment_type.TEXT)\n            allure.attach(f\"Cards  Created Successfully: {cards_created_successfully}\",\n                          name=\"card created successfully\",\n                          attachment_type=allure.attachment_type.TEXT)\n    \n            assert total_card_requests == records, f\"Mismatch in the total number of card requests. Expected: {records}, but got: {total_card_requests}.\"\n>           assert cards_created_successfully == GiftCardCreationWithSameMobileNumber.NSDL_GC, (\n                f\"Unexpected number of fino personalized cards created successfully. Expected: 30, but got: {cards_created_successfully}.\"\n            )\nE           AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.\n\npages/gcBulk/digitalGc/DigitalCardCreationInBulk.py:582: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1738691690677,"stop":1738691690705,"duration":28},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"env_config","time":{"start":1738691690718,"stop":1738691690718,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"finoObject","time":{"start":1738694997256,"stop":1738694998053,"duration":797},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false}],"testStage":{"description":"Verify the bulk creation of digital gift cards with the same mobile number in a negative scenario.","status":"failed","statusMessage":"AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.","statusTrace":"self = <TestCases.gcBulk.test_fino_digital_bulk_gc_card_creation.TestFinoDigitalBulkGiftCardCreation object at 0x102b089b0>\nfinoObject = <constant.fino.fino_constant.FinoConstant object at 0x102cd92e0>\n\n    @allure.description(\n        \"Verify the bulk creation of digital gift cards with the same mobile number in a negative scenario.\")\n    def test_to_verify_digital_card_creation_in_bulk_with_same_mob_with_negative_scenario(self, finoObject):\n>       DigitalGiftCardCreationInBulk.verify_zaggle_digital_card_creation_in_bulk_with_same_mob_negative_scenario(\n            finoObject.FINO_GC_CARD_PROGRAM_ID, \"fino_same_mob_neg\", 50, finoObject.ADMIN_LOGIN_TOKEN, Bank.FINO\n        )\n\nTestCases/gcBulk/test_fino_digital_bulk_gc_card_creation.py:101: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages/gcBulk/digitalGc/DigitalCardCreationInBulk.py:365: in verify_zaggle_digital_card_creation_in_bulk_with_same_mob_negative_scenario\n    DigitalGiftCardCreationInBulk.verify_fino_failed_bulk_card_creation_for_same_mob_number(response,\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nresponse = <Response [200]>, records = 50, token = 'rYkNz0AQn4nyMMjDGw33T9pliEE5I7tmfKnow5L2bJ8='\n\n    @staticmethod\n    @allure.step(\"verify failed bulk upload for same mob number \")\n    def verify_fino_failed_bulk_card_creation_for_same_mob_number(response, records, token):\n        with allure.step(\"Verify bulk upload constraints and validate results\"):\n            CommonMethod.timeTakenForBulkUpload(records)\n    \n            bulkUploadId = response.json().get(\"bulkUploadId\")\n            allure.attach(bulkUploadId, name=\"Bulk Upload ID\", attachment_type=allure.attachment_type.TEXT)\n    \n            responseOfBulkUpload = BulkUploadDetails.get_bulk_upload_details(bulkUploadId, token)\n            data = responseOfBulkUpload.json()\n            allure.attach(str(data), name=\"Bulk Upload Details\", attachment_type=allure.attachment_type.JSON)\n    \n            status = data.get(\"status\")\n    \n            # assert status.lower() == \"failed\", \"Bulk upload should fail,whole file will failed for zaggle\"\n    \n            # else:\n            assert status.lower() == \"completed\", \"As 30 card will created successfully with same mob so status will be completed, This is for fino rest of all failed\"\n    \n            total_card_requests = data.get(\"totalCardsRequested\")\n            cards_not_created = data.get(\"cardsNotCreated\")\n            cards_created_successfully = data.get(\"cardsCreatedSuccessfully\")\n            allure.attach(f\"Total Cards Requested: {total_card_requests}\", name=\"Total Cards Requested\",\n                          attachment_type=allure.attachment_type.TEXT)\n            allure.attach(f\"Cards Not Created: {cards_not_created}\", name=\"Cards Not Created\",\n                          attachment_type=allure.attachment_type.TEXT)\n            allure.attach(f\"Cards  Created Successfully: {cards_created_successfully}\",\n                          name=\"card created successfully\",\n                          attachment_type=allure.attachment_type.TEXT)\n    \n            assert total_card_requests == records, f\"Mismatch in the total number of card requests. Expected: {records}, but got: {total_card_requests}.\"\n>           assert cards_created_successfully == GiftCardCreationWithSameMobileNumber.NSDL_GC, (\n                f\"Unexpected number of fino personalized cards created successfully. Expected: 30, but got: {cards_created_successfully}.\"\n            )\nE           AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.\n\npages/gcBulk/digitalGc/DigitalCardCreationInBulk.py:582: AssertionError","steps":[{"name":"Fetch card program details for card program ID: 250204142857543ID1CP0817422","time":{"start":1738695199678,"stop":1738695199866,"duration":188},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"Generate bulk file for digital gift card creation: fino_same_mob_neg with 50 records with same mobile number","time":{"start":1738695199866,"stop":1738695200124,"duration":258},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"Send payload for bulk upload with file path: /Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/constant/BULK_CARD_UPLOAD_FILES/Digital/fino_same_mob_neg_20250205002319.csv","time":{"start":1738695200124,"stop":1738695200371,"duration":247},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"hasContent":false,"shouldDisplayMessage":false,"attachmentsCount":0,"attachmentStep":false},{"name":"verify bulk upload, bulk_upload_id: 250204185320361ID1BULK2362730","time":{"start":1738695200371,"stop":1738695320617,"duration":120246},"status":"failed","statusMessage":"AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.\n","statusTrace":"  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/pages/gcBulk/digitalGc/DigitalCardCreationInBulk.py\", line 365, in verify_zaggle_digital_card_creation_in_bulk_with_same_mob_negative_scenario\n    DigitalGiftCardCreationInBulk.verify_fino_failed_bulk_card_creation_for_same_mob_number(response,\n  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/.venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/pages/gcBulk/digitalGc/DigitalCardCreationInBulk.py\", line 582, in verify_fino_failed_bulk_card_creation_for_same_mob_number\n    assert cards_created_successfully == GiftCardCreationWithSameMobileNumber.NSDL_GC, (\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[{"name":"verify failed bulk upload for same mob number ","time":{"start":1738695200372,"stop":1738695320616,"duration":120244},"status":"failed","statusMessage":"AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.\n","statusTrace":"  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/.venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/pages/gcBulk/digitalGc/DigitalCardCreationInBulk.py\", line 582, in verify_fino_failed_bulk_card_creation_for_same_mob_number\n    assert cards_created_successfully == GiftCardCreationWithSameMobileNumber.NSDL_GC, (\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[{"name":"Verify bulk upload constraints and validate results","time":{"start":1738695200372,"stop":1738695320614,"duration":120242},"status":"failed","statusMessage":"AssertionError: Unexpected number of fino personalized cards created successfully. Expected: 30, but got: 30.\n","statusTrace":"  File \"/Users/shashikushwaha/Documents/workspace/c91_api_automation_ppi/pages/gcBulk/digitalGc/DigitalCardCreationInBulk.py\", line 582, in verify_fino_failed_bulk_card_creation_for_same_mob_number\n    assert cards_created_successfully == GiftCardCreationWithSameMobileNumber.NSDL_GC, (\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","steps":[],"attachments":[{"uid":"a77db397252fb21c","name":"Bulk Upload ID","source":"a77db397252fb21c.txt","type":"text/plain","size":29},{"uid":"a42adafc8fa648e3","name":"Bulk Upload Details","source":"a42adafc8fa648e3.json","type":"application/json","size":1126},{"uid":"60f68df7cc35f801","name":"Total Cards Requested","source":"60f68df7cc35f801.txt","type":"text/plain","size":25},{"uid":"1f112e2b659174ba","name":"Cards Not Created","source":"1f112e2b659174ba.txt","type":"text/plain","size":21},{"uid":"c55c68ad767c5aab","name":"card created successfully","source":"c55c68ad767c5aab.txt","type":"text/plain","size":31}],"parameters":[],"stepsCount":0,"hasContent":true,"shouldDisplayMessage":true,"attachmentsCount":5,"attachmentStep":false}],"attachments":[],"parameters":[{"name":"response","value":"<Response [200]>"},{"name":"records","value":"50"},{"name":"token","value":"'rYkNz0AQn4nyMMjDGw33T9pliEE5I7tmfKnow5L2bJ8='"}],"stepsCount":1,"hasContent":true,"shouldDisplayMessage":false,"attachmentsCount":5,"attachmentStep":false}],"attachments":[],"parameters":[],"stepsCount":2,"hasContent":true,"shouldDisplayMessage":false,"attachmentsCount":5,"attachmentStep":false}],"attachments":[{"uid":"c7a3da6e5b66c789","name":"log","source":"c7a3da6e5b66c789.txt","type":"text/plain","size":170}],"parameters":[],"stepsCount":6,"hasContent":true,"shouldDisplayMessage":true,"attachmentsCount":6,"attachmentStep":false},"afterStages":[],"labels":[{"name":"parentSuite","value":"TestCases.gcBulk"},{"name":"suite","value":"test_fino_digital_bulk_gc_card_creation"},{"name":"subSuite","value":"TestFinoDigitalBulkGiftCardCreation"},{"name":"host","value":"MacBookAir.bbrouter"},{"name":"thread","value":"49510-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"TestCases.gcBulk.test_fino_digital_bulk_gc_card_creation"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"3f804b9397f2e684.json","parameterValues":[]}